classes/UpgradeSelfCheck.php

```diff
@@ -400,9 +401,29 @@ private function checkPhpVersionNeedsUpgrade()
      */
     private function checkShopIsDeactivated()
     {
-        return
-            !Configuration::get('PS_SHOP_ENABLE')
-            || (isset($_SERVER['HTTP_HOST']) && in_array($_SERVER['HTTP_HOST'], ['127.0.0.1', 'localhost', '[::1]']));
+        // always return true in localhost
+        if (in_array($this->getRemoteAddr(), ['127.0.0.1', 'localhost', '[::1]', '::1'])) {
```

Comment:
@PierreRambaud You can bypass the `X-Forwarded-For` but not the `REMOTE_ADDR` I think? So, if the server is well configured, I think we're safe as the `X-Forwarded-For` will not be take into account if the `REMOTE_ADDR` is not a local address. And if the `REMOTE_ADDR` is a local address, only the proxy should set the `X-Forwarded-For` header. Or am I missing something? 😅

Reply:
Look what `getRemoteAddr` is doing, it's easy to bypass ^^

Reply:
These values should take the maintenance IP instead of localhost only

Reply:
I tried to see how `getRemoteAddr` can be bypassed, but to me, as long as `REMOTE_ADDR` can't be spoofed, I don't see how you can bypass it. Am I missing something?

Reply:
`getRemoteAddr` can rely on  `X-Forwarded-For` so it can be spoofed. A lot of host provider are using nginx as a reverse proxy. Most of the time you can easily spoof it. But anyway, it was just a comment.
I was more in favor of configuring it in the BO like the maintenance page than using string (or `const char*`, didn't remember how it's call in PHP :eyes: )

Reply:
I know about `X-Forwarded-For`, but `REMOTE_ADDR` can't be spoofed AFAIK. So if we check that `REMOTE_ADDR` is a local IP address, it means that `X-Forwarded-For` as been set by the local address and then can be trusted.